### ISSUE TITLE
Azure Pipeline failing at react-native-windows/Directory.Build.targets(78,5) with 'Syntax of Time-Parameter is wrong', Local build works

### ISSUE THREAD

**Issue by creambyemute** [microsoft/react-native-windows#7866]
Hello there,

We're close to releasing our RN-Windows App to the wild and are setting up the ci/cd pipeline with azure.
Unfortunately we were not able to use the MS-Hosted agents due to 10gb size limit (Idk why, my local build folder is only 2.6gb big) so I setup a Self-Hosted Agent on my working machine.

Whenever I build via the pipeline it succeeds almost to the end until the step `CleanUpPrecompHeaders ` and aborts there with:
```
_GenerateProjectPriFileCore:
 C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x64\MakePri.exe New -ProjectRoot C:\agent\_work\1\s\node_modules\react-native-windows\Microsoft.ReactNative\ -ConfigXml C:\agent\_work\1\s\node_modules\react-native-windows\\build\x64\Release\Microsoft.ReactNative\priconfig.xml -OutputFile C:\agent\_work\1\s\node_modules\react-native-windows\target\x64\Release\Microsoft.ReactNative\Microsoft.ReactNative.pri -IndexName Microsoft.ReactNative -Verbose -Overwrite  
CleanUpPrecompHeaders:
  The file "C:\agent\_work\1\s\node_modules\react-native-windows\build\x64\Release\Microsoft.ReactNative\pch.pch" is being deleted.
##[error]node_modules\react-native-windows\Directory.Build.targets(78,5): Error MSB3376: The syntax of the Time parameter is incorrect. Die Zeichenfolge wurde nicht als gültiges DateTime erkannt.
```

That's a generated datetime and I'm pretty clueless on what's going wrong here, the file states though that: `As a cppwinrt project, .pch is very big, and delete it in pipeline after build
    Instead of outright deleting the PCHs, we want to trick the "project freshness detector" by
    emitting empty files that look suspiciously like the PCHs it's expecting.
    It's of utmost importance that their timestamps match up`

So it seems they do not match up, but why? A Local build, not via the azure pipeline agent works flawlessly.

**Comment by chrisglein** [comment]
> We're close to releasing our RN-Windows App to the wild

Very exciting!

> Directory.Build.targets(78,5): Error MSB3376

There is a call to some time APIs. The guess here is that there's something locale sensitive that's mismatching between the various calls. We'll look into it.

There's this comment nearby:

> As a C++/WinRT project, .pch is very big, and delete it in pipeline after build
    Instead of outright deleting the PCHs, we want to trick the "project freshness detector" by
    emitting empty files that look suspiciously like the PCHs it's expecting.
    It's of utmost importance that their timestamps match up. -->

This was likely part of the protection against big PCH files. @asklar has been trying to drive feedback and clarity on this.

**Comment by asklar** [comment]
As for the Time parameter error, this is likely a bug on MSBuild's [Touch task](https://docs.microsoft.com/en-us/visualstudio/msbuild/touch-task) where it is not understanding the time format given back in your system locale; I think it should be possible to remove this task from the directory.build.targets to get you unblocked. Could you file a bug against msbuild? https://github.com/dotnet/msbuild

**Comment by DHowett** [comment]
As the person who authored that rule for the Terminal project (fancy meeting it here!), I apologize :wink:

**Comment by creambyemute** [comment]
I have tried different language + region settings now and couldn't find one that works:

* US Display Language + US Region Formatting
* US Display Language + DE-CH Region Formatting
* DE-CH Display Language + DE-CH Region Formatting
* DE-CH Display Language + US Region Formatting

@asklar Shall I still open an issue in the MSBuild repo?

**Comment by asklar** [comment]
Good to know you're unblocked @creambyemute - we're taking a change that removes the logic from RNW so it should be fixed in 0.65.
